keps/sig-node/3675-resource-plugin-manager/README.md

```diff
@@ -0,0 +1,907 @@
+be a simple way to allow users to meet their specific experiments.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+As the feature is stand-alone and behind a feature gate, this should not
+impact standard Kubelet behavior.  In the event someone is using the CCI
+Drivers, the alpha version will default to standard best effort QoS until
+the CCI Driver is deployed.  After deployment, the management will be reliant on
+the CCI Driver implementation.
+
+## Design Details
+
+#### Compute Specification Option
+
+The CCI model combined with some of the capabilities introduced in the Dynamic 
+Resource Allocation (DRA) KEP [3], offers the ability to transition compute 
+resource allocation behavior from being a cluster-admin dominated configuration
+to one that allows users with precise compute requirements to articulate the
+compute attributes that they need. 
+
+In several domains such Streaming, Telco, HPC, and AI/ML, users require
+fine-grained compute resource control to maximize their key performance 
+indicators.  They need an API option that can be made available through
+upstream Kubernetes components and specific compute resource allocation 
+plugins.
+
+##### Example 1: Attributed-based resource specification
+
+One realization of such an API can be achieved by an attribute-based 
+mechanism for compute resources. The attribute-based compute resource 
+request model is a step towards improved semantic workload portability 
+that can be tried in a progressive manner with CCI. 
+
+In the following, there is an example of a novel approach that could be 
+fulfilled with the CCI architecture. This example presents a set of
+per-core attributes in a config-map form that enable precision beyond a policy-only based approach: 
+
+    # A resource request which consists of:
+    # 2 exclusive cores (no other processes running on them)
+    # 	* the cores shall be siblings
+    # 	* they shall run at 2.1 ghz
+    # 	* application assigned has the highest priority on the core
+    # 	* there are no IRQs on the core
+    # 6 exclusive cores(no other processes running on them)
+    # 4 shared cores(other processes can run on them) 
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: example-compute-claim-parameters
+      namespace: default
+    data:
+      cores: |
+    # number of core type 1
+    2
+    # number of core type 2
+    6
+    # number of core type 3
+    4
+      compute_attributes: |
+        # Attributes of core type 1
+        exclusive, smt-sibling-required, 2.1Ghz, 1.0 priority, no-irq
+        # Attributes of core type 2
+        exclusive
+        # Attributes of core type 3
+        shared
```

Comment:
How this ConfigMap is going to be used and by which component (Kubelet, Scheduler, CCI Plugin, etc)?

Reply:
remains an open question, such API changes will not be covered in alpha. We wanted to point out one possibility for more fine-grained resource configuration, it is a motivation example.

Reply:
I'd still prefer to understand how that would work, at least generally. I'm not sure I support deferring this explanation to the future as it's an important design decision and should be made **before** this KEP is approved.

Reply:
I _strongly_ agree with @bart0sh here.

Reply:
we extended the example list and set a reasonable path for alpha versions , we outlined several options for post-alpha.